FirebaseStorage/Sources/Internal/StorageDeleteTask.swift

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+import FirebaseStorageInternal
+#if COCOAPODS
+  import GTMSessionFetcher
+#else
+  import GTMSessionFetcherCore
+#endif
+
+/**
+ * Task which provides the ability to delete an object in Firebase Storage.
+ */
+internal class StorageDeleteTask: StorageTask, StorageTaskManagement {
+  private var fetcher: GTMSessionFetcher?
+  private var fetcherCompletion: ((Data?, NSError?) -> Void)?
+  private var completion: ((_ error: Error?) -> Void)?
+
+  internal init(reference: FIRIMPLStorageReference,
+                fetcherService: GTMSessionFetcherService,
+                queue: DispatchQueue,
+                completion: ((_: Error?) -> Void)?) {
+    super.init(reference: reference, service: fetcherService, queue: queue)
+    self.completion = completion
+  }
+
+  deinit {
+    self.fetcher?.stopFetching()
+  }
+
+  /**
+   * Prepares a task and begins execution.
+   */
+  internal func enqueue() {
+    weak var weakSelf = self
+    DispatchQueue.global(qos: .background).async {
+      guard let strongSelf = weakSelf else {
+        return
+      }
```

Comment:
Would it be ever so slightly cleaner to use the closure capture list syntax (not sure what the actual name is)? I.e.,
```
DispatchQueue.global(qos: .background).async { [weak self] in
  guard let strongSelf = self else { return }
```
I _think_ that would ensure that any weak references to self below would end up being optional, which could be a nice safety check.

Reply:
Good thought and I learned something from investigating. However, the self references in the completion block need to be retained to make sure that the completion block always gets called.